caps/eth.md

```diff
@@ -0,0 +1,223 @@
+faster than re-executing all transactions but comes at the expense of some security.
+
+State synchronization typically proceeds by downloading the chain of block headers,
+verifying their proof-of-work values. Block bodies are requested as in the Basic Chain
+Synchronization section but block transactions aren't executed. Instead, the client picks
+a block near the head of the chain and downloads merkle tree nodes and contract code
+incrementally by requesting the root node, it's children, grandchildren, ... until the
+entire tree is synchronized.
+
+## Protocol Messages
+
+### Status (0x00)
+
+`[protocolVersion: P, networkId: P, td: P, bestHash: B_32, genesisHash: B_32, number: P]`
+
+Inform a peer of its current ethereum state. This message should be sent _after_ the
+initial handshake and _prior_ to any **ethereum** related messages.
+
+* `protocolVersion`: the current protocol version, 63
+* `networkId`: Integer identifying the blockchain, see table below
+* `td`: total difficulty of the best chain. Integer, as found in block header.
+* `bestHash`: The hash of the best (i.e. highest TD) known block.
+* `genesisHash`: The hash of the Genesis block.
+* `number`: The block number of the latest block in the chain.
+
+This table lists common Network IDs and their corresponding networks. Other IDs exist
+which aren't listed, i.e. clients should not require that any particular network ID is
+used.
+
+| ID | chain                              |
+|----|------------------------------------|
+| 0  | Olympic (disused)                  |
+| 1  | Frontier (now mainnet)             |
+| 2  | Morden (disused)                   |
+| 3  | Ropsten (current PoW testnet)      |
+| 4  | [Rinkeby](https://www.rinkeby.io/) |
+
+### NewBlockHashes (0x01)
+
+`[[hash_0: B_32, number_0: P], [hash_1: B_32, number_1: P], ...]`
+
+Specify one or more new blocks which have appeared on the network. To be maximally
+helpful, nodes should inform peers of all blocks that they may not be aware of. Including
+hashes that the sending peer could reasonably be considered to know (due to the fact they
+were previously informed of because that node has itself advertised knowledge of the
+hashes through NewBlockHashes) is considered bad form, and may reduce the reputation of
+the sending node. Including hashes that the sending node later refuses to honour with a
+proceeding GetBlockHeaders message is considered bad form, and may reduce the reputation
+of the sending node.
+
+### Transactions (0x02)
+
+`[[nonce: P, receivingAddress: B_20, value: P, ...], ...]`
+
+Specify (a) transaction(s) that the peer should make sure is included on its transaction
+queue. The items in the list (following the first item `0x12`) are transactions in the
+format described in the main Ethereum specification. Nodes must not resend the same
+transaction to a peer in the same session. This packet must contain at least one (new)
```

Comment:
Nodes must also not send a transaction back to that same peer if it was learned from that peer?